dietpi/dietpi-software

```diff
@@ -3508,6 +3517,74 @@ _EOF_
 
 		fi
 
+		software_id=200 # DietPi Dashboard
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Installing
+
+			arch=
+			if (( $G_HW_ARCH == 10 )); then
+				arch='amd64'
+			elif (( $G_HW_ARCH == 3 )); then
+				arch='armv8'
+			elif (( $G_HW_ARCH == 2 )); then
+				arch='armv7'
+			else
+				arch='armv6'
+			fi
+
+			G_WHIP_MENU_ARRAY=(
+					'Stable' ': Use a release version of the DietPi Dashboard.'
+					'Nightly' ': Use an unstable version of the DietPi Dashboard.'
+			)
+
+			local choice_required=
+			while :
+			do
+
+				G_WHIP_MENU "${choice_required}Please choose which version of DietPi Dashboard should be installed." || { choice_required='[ERROR] A choice is required to finish the DietPi Dashboard install.\n\n'; continue; }
+				break
+
+			done
+
+			mkdir -p /opt/dietpi-dashboard
+
+			if [[ $G_WHIP_RETURNED_VALUE == 'Stable' ]]
+			then
+				Download_Install "$(curl -sSfL 'https://api.github.com/repos/ravenclaw900/dietpi-dashboard/releases/latest' | mawk -F\" "/\"browser_download_url\": \".*dietpi-dashboard-$arch\"/{print \$4}")" /opt/dietpi-dashboard/dietpi-dashboard
+			elif [[ $G_WHIP_RETURNED_VALUE == 'Nightly' ]]
+			then
+				no_check_url=1 Download_Install "https://nightly.link/ravenclaw900/DietPi-Dashboard/workflows/push-build/main/dietpi-dashboard-$arch.zip" dietpi-dashboard
+				G_EXEC mv dietpi-dashboard/dietpi_dashboard /opt/dietpi-dashboard/dietpi-dashboard
+			fi
+
+			G_EXEC chmod +x /opt/dietpi-dashboard/dietpi-dashboard
+
+			cat << _EOF_ > /opt/dietpi-dashboard/config.toml
```

Comment:
Shall we keep the convention to have any configs (and data) in `/mnt/dietpi_userdata/<name>`? But for now it is probably overkill with then a single file only in each dir, and it would require an additional option to define the config file path... We can move this any time later when more features which write data, or split config files are implemented.

But we shouldn't overwrite existing config files:
```sh
[[ -f '/opt/dietpi-dashboard/config.toml' ]] || cat << _EOF_ > /opt/dietpi-dashboard/config.toml
```